modules/analyze_plugin_example/src/Plugin/Analyze/Example.php

```diff
@@ -0,0 +1,225 @@
+   * of data should be restricted to the plugin's full report page.
+   */
+  public function renderSummary(EntityInterface $entity): array {
+
+    // This method must return a valid Drupal render array. This will be wrapped
+    // in a "fieldset" render array by the base module, and displayed on the
+    // summary page for the entity. This can be any valid render array: this
+    // example describes a table.
+    // First, you will have to implement your own methods to create your data
+    // from the provided entity as required.
+    $data = $this->doStuff($entity);
+
+    // Then that data will need to be formatted into an appropriate Drupal
+    // render array.
+    $build = [
+      '#type' => 'table',
+      '#header' => [['data' => 'Example', 'colspan' => 2, 'class' => ['header']]],
+      '#rows' => [],
+      '#attributes' => [
+        'class' => ['example-table'],
+        'style' => ['table-layout: fixed;'],
+      ],
+    ];
+
+    foreach ($data as $label => $value) {
+      $build['#rows'][] = [
+        'data' => [$label, $value],
+      ];
+    }
+
+    return $build;
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * By default, this method will return and empty array. If you intend to
+   * display additional metrics in a dedicated page, you can override this
+   * method to return a valid render array of data and it will be displayed by
+   * default at /<entity_type>/<entity_id>/analyze/<plugin_id>.
+   *
+   * If you do not wish to display a full report at the default URL, you MUST
+   * override the getFullReportUrl() method to return either an alternate URL or
+   * a NULL value to disable the default URL: once that is done, you can remove
+   * this method completely from your plugin. Failure to do so will result in an
+   * empty page being displayed at the default URL.
+   */
+  public function renderFullReport(EntityInterface $entity): array {
+
+    // Again, you will need to implement your own methods to obtain the data you
+    // wish to display on the full report page.
+    $data = $this->doStuff($entity);
+
+    // The default Full Report page will wrap any returned render array in a
+    // fieldset, but if you wish to add multiple arrays to your data you may
+    // want to wrap everything in your own fieldset for clarity.
+    $build = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Example Full Report'),
+    ];
+
+    // One you have your data, you can format it into a render array.
+    foreach ($data as $key => $values) {
+
+      // The base Analyze module provides an 'analyze_gauge' theming function
+      // which render a three point bar chart and display a value as a point
+      // somewhere on the scale. This uses a Drupal Single Directory Components
+      // to render the component, and so  can be used in your custom templates
+      // if required. See analyze_theme() and the /components folder for the
+      // implementation.
+      $build[$key] = [
```

Comment:
It would be nice if the example module worked so that I could see the gauge work for the first time but I doubt it will work because here it is spelled analyze_gauge and elsewhere "guage" as I mentioned before.

Reply:
Resolved in 45fc05ca4a3547af207ee74470da74e5490a32e2 and dce7d21939b0bae2da7cf80291dd7d1c051cee99